lib/galaxy/tool_util/linters/inputs.py

```diff
@@ -395,10 +395,10 @@ def lint_inputs(tool_source: "ToolSource", lint_ctx: "LintContext"):
                         f"Parameter [{param_name}]: attribute '{attrib}' is incompatible with validator of type '{vtype}'",
                         node=validator,
                     )
-            if vtype == "expression":
+            if vtype in ["expression", "regex"]:
                 if validator.text is None:
                     lint_ctx.error(
-                        f"Parameter [{param_name}]: expression validators are expected to contain text", node=validator
+                        f"Parameter [{param_name}]: {vtype} validators are expected to contain text", node=validator
```

Comment:
I'm a little surprised about the existing expression linting ... `expression` is just any python expression, so I think the `re.compile(validator.text)` validation is wrong ?

Reply:
Looking at the tool's source in IUC I thought exactly that, that `expression` here should have been `regex` from the beginning... but I was not sure how it was handled internally if those expression texts are then converted to matching regex somehow...

Should I just replace `expression` with `regex` and call this a bug?

Reply:
I'd maybe turn line 403 into `elif vtype == "regex":`, the non-empty text linting is probably correct.

I assume this wasn't noticed because most expressions probably compile fine into a regex ?

Reply:
> the non-empty text linting is probably correct

You mean a `regex` cannot be empty and must also have text right?

Reply:
both `regex` and `expresison` should not be empty, iow `if validator.text is None:` should be applied to both